pinot-plugins/pinot-stream-ingestion/pinot-kinesis/src/main/java/org/apache/pinot/plugin/stream/kinesis/server/KinesisDataProducer.java

```diff
@@ -0,0 +1,95 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.plugin.stream.kinesis.server;
+
+import java.net.URI;
+import java.util.Properties;
+import java.util.UUID;
+import org.apache.pinot.spi.stream.StreamDataProducer;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.core.SdkBytes;
+import software.amazon.awssdk.http.SdkHttpConfigurationOption;
+import software.amazon.awssdk.http.apache.ApacheSdkHttpService;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.kinesis.KinesisClient;
+import software.amazon.awssdk.services.kinesis.model.PutRecordRequest;
+import software.amazon.awssdk.services.kinesis.model.PutRecordResponse;
+import software.amazon.awssdk.utils.AttributeMap;
+
+
+public class KinesisDataProducer implements StreamDataProducer {
+  public static final String ENDPOINT = "endpoint";
+  public static final String REGION = "region";
+  public static final String ACCESS = "access";
+  public static final String SECRET = "secret";
+  public static final String DEFAULT_PORT = "4566";
+  public static final String DEFAULT_ENDPOINT = "http://localhost:4566";
+
+  private KinesisClient _kinesisClient;
+
+  @Override
+  public void init(Properties props) {
+    try {
+      if (props.containsKey(ENDPOINT)) {
+        String localStackKinesisEndpoint;
+        localStackKinesisEndpoint = props.getProperty(ENDPOINT, DEFAULT_ENDPOINT);
```

Comment:
rename this to just kinesisEndpoint? This should work for any forms of kinesis setup using endpoint right?

Reply:
done